innovation_sweet_spots/tests/test_query_categories.py

```diff
@@ -0,0 +1,139 @@
+from unittest import mock
+from innovation_sweet_spots.analysis.query_categories import (
+    query_gtr_categories,
+    query_cb_categories,
+)
+import pandas as pd
+
+
+def category_check_side_effect(*args, **kwargs):
+    if args[0] == "category_1":
+        return ["id_1", "id_3"]
+    elif args[0] == "category_2":
+        return ["id_2"]
+
+
+@mock.patch("innovation_sweet_spots.analysis.query_categories.initialise_gtr_id_table")
+@mock.patch(
+    "innovation_sweet_spots.analysis.query_categories.is_gtr_project_in_category"
+)
+def test_query_gtr_categories(
+    mock_is_gtr_project_in_category, mock_initialise_gtr_id_table
+):
+
+    mock_is_gtr_project_in_category.return_value = []
+    mock_is_gtr_project_in_category.side_effect = category_check_side_effect
+    mock_initialise_gtr_id_table.return_value = pd.DataFrame(
+        {"id": ["id_1", "id_2", "id_3"]}
+    )
```

Comment:
Here, you can move the return values and side effects into `@mock.patch`
```suggestion
@mock.patch(
    "innovation_sweet_spots.analysis.query_categories.initialise_gtr_id_table",
    return_value=pd.DataFrame({"id": ["id_1", "id_2", "id_3"]}),
)
@mock.patch(
    "innovation_sweet_spots.analysis.query_categories.is_gtr_project_in_category",
    return_value=[],
    side_effect=category_check_side_effect,
)
def test_query_gtr_categories(
    mock_is_gtr_project_in_category, mock_initialise_gtr_id_table
):
```

Reply:
Nice, that is neater - thank you!